Synergy/Frontend/src/app/features/home-page/components/home-page/home-page.component.ts

```diff
@@ -1,36 +1,43 @@
 import {Component, OnDestroy} from '@angular/core';
-import {FirebaseAuthService} from "../../../../services/auth/firebase-auth.service";
-import firebase from "firebase/compat";
-import User = firebase.User;
-import {GetUserPermissionsService} from "../../../../services/api-requests/users/get-user-permissions.service";
-import {Router} from "@angular/router";
+import {GetUserService} from "../../../../services/api-requests/users/get-user.service";
+import {Select, Store} from "@ngxs/store";
+import {UserStateModel} from "../../../../store/states/user.state";
+import {Observable} from "rxjs";
+import {UserActions} from "../../../../store/actions/user.action";
+import {GetTeamsService} from "../../../../services/api-requests/teams/get-teams.service";
+import {TeamsStateModel} from "../../../../store/states/teams.state";
+import {Team} from "../../../../store/actions/teams.action";
+
+interface LoggedInUser {
+  userId: string,
+  role: string;
+  permissions: string[],
+  teams: string[]
+}
+
 
 @Component({
   selector: 'app-home-page',
   templateUrl: './home-page.component.html',
   styleUrl: './home-page.component.css'
 })
-export class HomePageComponent implements OnDestroy{
-  user: User | null = null;
+export class HomePageComponent implements OnDestroy {
+  userId: string = '';
   permissions: string[] = [];
-  constructor(private firebaseAuthService: FirebaseAuthService, private getPermissionService: GetUserPermissionsService, private router: Router) {
-  }
-  userSubscription = this.firebaseAuthService.currentUser$.subscribe(user => {
-    this.user = user;
-    if (user)
-      this.getPermissionService.getUserPermissions(user?.uid);
-  })
-  permissionsSubscription = this.getPermissionService.permissions$.subscribe(permissions => {
-    this.permissions = permissions;
+  teams: Team[] = [];
+  @Select((state: { user: UserStateModel }) => state.user.loggedInUser)
+  loggedInUser$!: Observable<LoggedInUser>;
+  userSubscription = this.loggedInUser$.subscribe(user => {
```

Comment:
I don't think permissions are being used anymore so I think we could use the async pipe to fetch the user. This might be something we will have to pair on for me to properly explain.